src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -44,6 +54,237 @@ public ApiMethodTransformer() {
     this.initCodeTransformer = new InitCodeTransformer();
   }
 
+  public StaticApiMethodView generatePagedFlattenedMethod(
+      MethodTransformerContext context, ImmutableList<Field> fields) {
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(context.getNamer().getApiMethodName(context.getMethod()));
+    setListMethodFields(context, methodViewBuilder);
+    setFlattenedMethodFields(context, fields, methodViewBuilder);
+
+    return methodViewBuilder.type(ApiMethodType.PagedFlattenedMethod).build();
+  }
+
+  public StaticApiMethodView generatePagedRequestObjectMethod(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(namer.getApiMethodName(context.getMethod()));
+    setListMethodFields(context, methodViewBuilder);
+    setRequestObjectMethodFields(
+        context, namer.getPagedCallableMethodName(context.getMethod()), methodViewBuilder);
+
+    return methodViewBuilder.type(ApiMethodType.PagedRequestObjectMethod).build();
+  }
+
+  public StaticApiMethodView generatePagedCallableMethod(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    StaticApiMethodView.Builder methodViewBuilder = StaticApiMethodView.newBuilder();
+
+    setCommonFields(context, methodViewBuilder);
+    methodViewBuilder.name(namer.getPagedCallableMethodName(context.getMethod()));
+    setListMethodFields(context, methodViewBuilder);
+    setCallableMethodFields(
+        context, namer.getPagedCallableName(context.getMethod()), methodViewBuilder);
+
+    return methodViewBuilder.type(ApiMethodType.PagedCallableMethod).build();
+  }
+
+  public StaticApiMethodView generateUnpagedListCallableMethod(MethodTransformerContext context) {
```

Comment:
So it turns out that you were the one who introduced this terminology:

https://github.com/googleapis/toolkit/commit/a9805c835f81f39735814724c3b3a9ee03277be0#diff-e184ca12b1548f2c4cc6af9e753e70dc

In defense of the naming scheme, paged calls are always on list methods, whereas the term "unpaged" could theoretically be applied to both list and non-list methods.